oak/server/rust/oak_runtime/src/lib.rs

```diff
@@ -424,6 +437,44 @@ impl Runtime {
     pub fn gather_metrics(&self) -> Vec<MetricFamily> {
         self.metrics_data.gather()
     }
+
+    /// Check Wasm module signatures.
+    /// Since such signatures are optional - tries to find corresponding signatures by module names
+    /// and verifies them if found.
+    pub(crate) fn verify_module_signatures(&self) -> Result<(), OakStatus> {
+        for (name, module) in &self.application_configuration.wasm_modules {
+            let module_hash = get_hash(&module);
+            // Get signature by module name.
+            if let Some(signatures) = self.signature_table.get(&module_hash) {
+                for signature_item in signatures.iter() {
+                    let public_key = &signature_item.public_key;
+                    let signature = &signature_item.signature;
+                    let data_reader = std::io::Cursor::new(module);
+
+                    // Output status information to `stderr`.
+                    // https://docs.rs/minisign/0.5.20/minisign/fn.verify.html#arguments
+                    const STATUS_TO_STDERR: bool = false;
+                    // Output a copy of the data to `stdout`.
+                    const DATA_TO_STDOUT: bool = false;
+                    verify(
+                        public_key,
+                        signature,
+                        data_reader,
+                        STATUS_TO_STDERR,
+                        DATA_TO_STDOUT,
+                    )
+                    .map_err(|error| {
+                        error!(
+                            "Wasm module signature verification failed for {}: {}",
+                            name, error
+                        );
+                        OakStatus::ErrInvalidArgs
```

Comment:
Not sure we want to fail this way long term; it's fine for now, but worth thinking about it in a future PR.

Reply:
Do you mean `ErrInvalidArgs` or just erroring with an `OakStatus`?

Reply:
Either way, don't have a strong opinion yet.